docs/docs/learn/concepts/storage/notes.md

```diff
@@ -0,0 +1,67 @@
+- "Note" types are part of aztec.nr, a framework that facilitates use of Aztec's different storage trees to achieve private variables. Note: there are more uses beyond this.
+
+This article will restrict its focus to how private variables are implemented with Notes and relevant storage trees. NB: these components can be used in a variety of ways to implement different attributes such as: singletons, unique value, or private sharing.
+
+#### Side-note about execution
+Under the hood, the Aztec protocol handles some important details around public and private function calls. Notably around calls between them and relevant delays to avoid race conditions around state reads/writes (more [here](../pxe/acir_simulator#simulating-functions)).
+Whilst it just works for us under the hood, a detailed explanation of the transaction lifecycle can be found [here](../transactions#simple-example-of-the-private-transaction-lifecycle).
+
+## Private state variables in Aztec
+State variables in an Aztec contract are defined inside a struct specifically named `Storage`, and must satisfy the [Note Interface](https://github.com/AztecProtocol/aztec-packages/tree/#include_aztec_version/noir-projects/aztec-nr/aztec/src/note/note_interface.nr) and contain a [Note header](https://github.com/AztecProtocol/aztec-packages/tree/#include_aztec_version/noir-projects/aztec-nr/aztec/src/note/note_header.nr).
+
+The Note header struct contains the contract address which the value is effectively siloed to, a nonce to ensure unique Note hashes, and a storage "slot" (or ID) to associate multiple notes.
+
+A couple of things to unpack here:
+
+#### Storage "slot"
+Storage slots are more literal for public storage, a place where a value is stored. For private storage, a storage slot is logical (more [here](./storage_slots#private-state-slots---slots-arent-real)).
+
+#### Silos
+The address of the contract is included in a Note's data to ensure contracts don't arrive at the same hash with an identical variable. This is handled in the protocol's execution.
+
+### Note types
+There is more than one Note type, but the `Set` type is most useful for private variables. There are also `Singleton` and `ImmutableSingleton` types.
+
+Furthermore, notes can be completely custom types, storing any value or set of values that are desired by an application.
+
+### Initialisation
+Private state variables are encrypted and stored locally when the contract is created. Depending on the application, values may be privately shared by the creator with others via encrypted logs onchain.
+A hash of a note is stored in the append-only note hash tree so as to prove existence of the current state of the note.
+
+#### Note Hash Tree
+By virtue of being append only, notes are not edited. If two transactions amend a private value, multiple notes will be inserted into the tree. The header will contain the same logical storage slot.
```

Comment:
maybe mention nullifier tree here? when a note is updated, both a nullifier and a note is created

Reply:
Nullifying is introduced in the next section.